compiler/rustc_mir_transform/src/abort_unwinding_calls.rs

```diff
@@ -107,31 +102,14 @@ impl<'tcx> MirPass<'tcx> for AbortUnwindingCalls {
             }
         }
 
-        // For call instructions which need to be terminated, we insert a
-        // singular basic block which simply terminates, and then configure the
-        // `cleanup` attribute for all calls we found to this basic block we
-        // insert which means that any unwinding that happens in the functions
-        // will force an abort of the process.
-        if !calls_to_terminate.is_empty() {
-            let bb = BasicBlockData {
-                statements: Vec::new(),
-                is_cleanup: true,
-                terminator: Some(Terminator {
-                    source_info: SourceInfo::outermost(body.span),
-                    kind: TerminatorKind::Abort,
-                }),
-            };
-            let abort_bb = body.basic_blocks_mut().push(bb);
-
-            for bb in calls_to_terminate {
-                let cleanup = body.basic_blocks_mut()[bb].terminator_mut().unwind_mut().unwrap();
-                *cleanup = Some(abort_bb);
-            }
+        for id in calls_to_terminate {
+            let cleanup = body.basic_blocks_mut()[id].terminator_mut().unwind_mut().unwrap();
+            *cleanup = UnwindAction::Terminate;
         }
 
         for id in cleanups_to_remove {
             let cleanup = body.basic_blocks_mut()[id].terminator_mut().unwind_mut().unwrap();
-            *cleanup = None;
+            *cleanup = UnwindAction::Unreachable;
```

Comment:
This would currently be unsound as a result of the FIXME in on `layout::fn_can_unwind`, see [here](https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/Unwinding.20into.20a.20panic.3Dabort.20crate) for an example that would miscompile

Reply:
I don't see how that's unsound. Mixing `-Cpanic=abort` and `-Cpanic=unwind` crates will result in `-Cpanic=abort`. There used to be an unsoundness but I have already fixed that in #97235.

Reply:
Nvm, see follow up discussion